src/cycles/johnson.jl

```diff
@@ -37,7 +37,7 @@ A more efficient version is possible.
 """
 function maxsimplecycles end
 @traitfn function maxsimplecycles(dg::::IsDirected, byscc::Bool=true)
-    c = 0
+    c::BigInt = zero(BigInt)

```

Comment:
This will kill performance. Why do we need `BigInt` here?

Reply:
Because the function `ncycles_n_i` returns a `BigInt`. The maximum number of cycles grows very fast, so a `Int` is probably not enough in many cases anyway.

Reply:
Wow. OK. Why would someone need to know that there are more than 2^64 cycles, and is that check going to terminate in any reasonable time?

Reply:
Maybe so that person knows, that it might be futile to try to count all cycles.
`maxsimplecycles` just gives a quick upper bound.  So it terminates quite fast, for example
```
julia> g = CompleteDiGraph(100)
{100, 9900} directed simple Int64 graph

julia> @time maxsimplecycles(g)
  0.000409 seconds (2.34 k allocations: 54.852 KiB)
25627587036766788968593938885807738338826187911012817324
```